uaclient/testing/fakes.py

```diff
@@ -34,3 +34,20 @@ def request_url(
         if isinstance(response, Exception):
             raise response
         return response, {"header1": ""}
+
+
+class FakeFile:
+    def __init__(self, content: str, name: str = "fakefile"):
+        self.content = content
+        self.cursor = 0
+        self.name = name
+
+    def read(self, size=None):
```

Comment:
Why can't `read` here just return the full content at once ?

Reply:
Because `yaml.load` calls `read` with the size parameter in a loop until `read` stops returning content (which I believe is a common pattern).
If we just do `return content`, then `yaml.load` never finishes because it thinks the file keeps returning more content